### ISSUE TITLE
Capturing Combat Drone uses 1 crew even though required crew is 0.

### ISSUE THREAD

**Issue by kyzentun** [endless-sky/endless-sky#2789]
Capturing navy combat drones for no reason and I noticed my crew ticking down for each one.

The lost crew member isn't reported killed, and since the combat drone has no bunks, the drone has 0 crew after the transfer.

So one unfortunate person is sent to capture the drone, reports a successful capture with no opposition, then promptly asphyxiates because there is no living area.

Maybe the transfer amount in Ship::WasCaptured should be 0 for a ship with no bunks?

**Comment by tehhowch** [comment]
It's actually a bit more interesting than that, and only happens if you are already low on crew.
```
// Transfer some crew over. Only transfer the bare minimum unless even that
	// is not possible, in which case, share evenly.
	int totalRequired = capturer->RequiredCrew() + RequiredCrew();
	int transfer = RequiredCrew();
	if(totalRequired > capturer->Crew())
		transfer = max(1, (capturer->Crew() * RequiredCrew()) / totalRequired);
	capturer->AddCrew(-transfer);
	AddCrew(transfer);
```
If you are low on crew *already*, then `totalRequired > capturer->Crew()` is true every time you capture something.

**Comment by kyzentun** [comment]
That explains why I didn't notice it when capturing 63 drones after the red vs blue fight in the "There Might Be Riots" storyline.

**Comment by tehhowch** [comment]
Indeed.

**Comment by endless-sky** [comment]
Fixed by merging #2790.
